anonymizer/OpenDiffix.Service/Program.fs

```diff
@@ -99,7 +104,22 @@ let handlePreview
         GROUP BY %s{String.join ", " [ 4 .. buckets.Length + 3 ]}
       """
 
-  let result = runQuery [ ledHook ] query inputPath anonParams
+  // We get a hold of the star bucket reference via side effects.
+  let mutable starBucket: Bucket option = None
+  let starBucketHook = AggregationHooks.StarBucket.hook (fun bucket -> starBucket <- Some bucket)
+
+  let result = runQuery [ ledHook; starBucketHook ] query inputPath anonParams
+
+  // Should be set once the query completes.
```

Comment:
this is just a runtime assertion, in lieu of proper tests, correct?

Reply:
What we have done is a little unorthodox. We use the callback trick to assign a mutable local variable (initially `None`) to our result. When unwrapping we have no choice but to fail in case it is `None`. No tests are needed because if it doesn't run _we will know_ from the crash.

I guess a test can assert the same thing and expect the star bucket to have specific stats.
In the tests file I would be happy if we can refactor in 2 modules similar to here, because the setup logic is becoming complex and knowing what is what helps a lot.


Currently `AggregationHooks.fs` is organized like this:

```
shared helpers

module Led =
  led helpers
  let hook = ....

module StarBucket
  star helpers
  let hook = ...
```